### ISSUE TITLE
Document how to use rules from code

### ISSUE THREAD

**Issue by petri** [vfaronov/turq#8]
I assume it must be somehow possible to define rules in a Python module and have turq use those. In other words, I want to store the rules and run turq using those. How can one do that?

**Comment by petri** [comment]
Ok this looks as simple as:

```
TurqHandler.code = """path('*').text("Hello World!")"""
TurqHandler.rules = parse_rules(TurqHandler.code)
```

**Comment by petri** [comment]
Ok it was simple to figure it out from code. But it would be nice if this was documented as part of package docs.

**Comment by petri** [comment]
Here's an example that you can use if you want (spawns turq server in a different process and then uses requests library to making an actual request to turq):

```
import BaseHTTPServer
from multiprocessing import Process

from turq import TurqHandler, parse_rules
import requests

HOST = "0.0.0.0"
PORT = 13085
RULES = """path('*').text("Hello World!")"""


def spawn_turq(host=HOST, port=PORT, rulecode=RULES):

    def serve(host, port, rulecode):
       TurqHandler.rules = parse_rules(rulecode)
       server = BaseHTTPServer.HTTPServer((host,port), TurqHandler)
       server.handle_request()

    p = Process(target=serve, args=(host, port, rulecode))
    p.start()
    return p


if __name__=="__main__":
   p = spawn_turq()
   r = requests.get("http://%s:%i" % (HOST, PORT))
   print r.text
   p.join()
```

**Comment by petri** [comment]
Of course it would be nice to add some log output, asserting that return text is hello world etc...  go ahead and add those if you like.

**Comment by petri** [comment]
Here's also a handy context manager that can be used to "mock" a http server ;)

``` python
@contextmanager
def mock_http_server(host, port, rulecode):
    def serve(host, port, rulecode):
       TurqHandler.rules = parse_rules(rulecode)
       server = BaseHTTPServer.HTTPServer((host,port), TurqHandler)
       server.handle_request()
    p = Process(target=serve, args=(host, port, rulecode))
    p.start()
    time.sleep(1) # wait a sec so the server has time to get up for sure
    yield
    p.join()
```

**Comment by vfaronov** [comment]
Thank you for the sample code.

Turq was intended specifically for interactive use, which is why I did nothing to provide a sane API. Several people have since requested a way to use Turq programmatically, so I definitely want to make this easier in the future. For the time being, I will document the present interface as you suggest. I’ll try to get around to this in a couple weeks.

By the way, have you seen [HTTPretty](https://github.com/gabrielfalcao/HTTPretty)? At the time I wrote Turq, I believed HTTPretty was the way to go for automated tests. I’ve since heard about [problems with HTTPretty](http://www.peterbe.com/plog/bye-bye-httpretty) so maybe there’s a place for Turq in automated scenarios as well.
